assembly/src/parsers/io_ops.rs

```diff
@@ -256,69 +276,137 @@ pub fn parse_loc_storew(op: &Token) -> Result<Node, ParsingError> {
 // HELPER FUNCTIONS
 // ================================================================================================
 
-fn parse_constants(op: &Token) -> Result<Vec<Felt>, ParsingError> {
-    let mut constants = Vec::new();
-    let param_idx = 1;
-    let param_count = op.num_parts() - param_idx;
+/// Parses a hexadecimal or decimal parameter values into a vector of u64 values valid for field
+/// element creation and returns appropriate instruction based on parsed values
+fn parse_param_list(op: &Token) -> Result<Node, ParsingError> {
+    let mut values = Vec::with_capacity(op.num_parts() - 1);
+    for (param_idx, &param_str) in op.parts().iter().enumerate().skip(1) {
+        let value = match param_str.strip_prefix("0x") {
+            Some(param_str) => parse_hex_value(op, param_str, param_idx)?,
+            None => parse_checked_param(op, param_idx, 0, Felt::MODULUS - 1)?,
+        };
+        values.push(value);
+    }
 
-    // for multiple input parameters, parse & push each one onto the stack in order, then return
-    if param_count > 1 {
-        for param_idx in param_idx..=param_count {
-            let value = parse_element_param(op, param_idx)?;
-            constants.push(value);
-        }
-        return Ok(constants);
+    build_push_many_instruction(values.iter())
+}
+
+/// Parses a hexadecimal parameter values in the form of strings of lenght 16 without delimeters
+/// into a vector of u64 values and returns appropriate instruction based on parsed values
```

Comment:
nit: I would change this to: "Parses a single hexadecimal parameter into multiple values and returns an appropriate push instruction node."